### DISCUSSION TITLE
System.InvalidOperationException  Message=Sequence contains no elements

### DISCUSSION THREAD

**Discussion by a3an-de** [SpiceSharp/SpiceSharpParser#158] [General]
Hi, I have taken the Parser Demo and replaced the netlist with a Spice list from the LTSpice astable.asc like this:

var netlistText = string.Join(Environment.NewLine,
                    "R1 n001 n002 2k",
                    "R2 n001 n003 2k",
                    "R3 n002 n004 101k",
                    "R4 n003 n005 100k",
                    "C1 n003 n004 .01µ",
                    "C2 n005 n002 .01µ",
                    "V1 n001 0 5",
                    "Q1 n003 n005 0 0 2n3904",
                    "Q2 n002 n004 0 0 2n3904",
                    ".MODEL 2n3904 npn(is=1e-14 vaf=100 bf=300 ikf=0.4 xtb=1.5 br=4 cjc=4e-12 cje=8e-12 rb=20 rc=0.1 re=0.1 tr=250e-9 tf=350e-12 itf=1 vtf=2 xtf=3 vceo=40 icrating=200m mfg=nxp)",
                    ".TRAN 25m startup",
                    ".END");

When executing I get the "Invalid  Operation Exception: Sequence contains no elements" at the following statement:
                var simulation = spiceSharpModel.Simulations.Single();

What is the Parser talking about ?

**Comment by marcin-golebiowski** [comment]
The parser doesn't support "startup" keyword for .TRAN. Because of that it ignores whole .TRAN statement
Please raise any issues you are having with the parser and I will be happy to help and solve.

Please find supported syntax here for .TRAN
http://bwrcs.eecs.berkeley.edu/Classes/IcBook/SPICE/UserGuide/analyses_fr.html

The parser needs to be improved. I will resume heavy work on it.

**Reply by a3an-de** [reply]
OK  Thanks.

I use 2 variations :  .TRAN 25m  and .TRAN 150m 
The first one got the sane exception
the second one got Simulations.Validation Failed Exception  'The simulation '#1 TRAN' has 1 rule violations'

What are the rules the Parser is talking about ?
Do I need to specify all 3 values as described in your wiki ?

**Reply by marcin-golebiowski** [reply]
According to https://ltwiki.org/LTspiceHelp/LTspiceHelp/_TRAN_Do_a_non_linear_transient_analysis.htm

.TRAN 25m is special to LtSpice. You need to specify at least 2 parameters at the moment 
e.g. ."TRAN `<step> <final>`"

**Comment by a3an-de** [comment]
Sorry, I made a mistake. My second variation was .TRAN 25m 150m
So I did specify 2 values  and got Simulations.Validation Failed Exception 'The simulation #1 TRAN' has 1 rule violations'
My question is what is the rule I am violating ?

**Reply by marcin-golebiowski** [reply]
Quick update for you. The exception is now from SpiceSharp core library:

![image](https://user-images.githubusercontent.com/115426/213815308-d6f7f49d-4be0-4cbc-8b09-441112f129db.png)

**Reply by marcin-golebiowski** [reply]
I know what is wrong. Please specify:

`            parser.Settings.Lexing.HasTitle = false;`

**Reply by marcin-golebiowski** [reply]
Please let me know if that helps in any way.

**Reply by a3an-de** [reply]
OK that fixed it. I am getting another problem...
This shows my ignorance ! I am not very firm in c# if at all. I need to talk to you about I am doing. Is there email address I can use ?
Here is the quick watch
![Screenshot (4)](https://user-images.githubusercontent.com/33397128/213819877-f29decbf-d5e4-4551-b2d3-2240c9efaf5c.png)

**Reply by marcin-golebiowski** [reply]
You need to add .SAVE statement to netlist and then in line 41 specify it by name.

**Reply by marcin-golebiowski** [reply]
Take a look here:
https://github.com/SpiceSharp/SpiceSharpParser/blob/main/src/SpiceSharpParser.IntegrationTests/DotStatements/SaveTests.cs

**Reply by marcin-golebiowski** [reply]
You can write me. My e-mail: `xormus@gmail.com`
